components/locale_canonicalizer/src/locale_canonicalizer.rs

```diff
@@ -285,18 +288,23 @@ impl LocaleCanonicalizer {
             // time of this loop was negligible.
             let mut matched = false;
             for rule in language_aliases.iter() {
-                if uts35_rule_matches(locale, &rule.0) {
-                    uts35_replacement(
-                        locale,
-                        !rule.0.language.is_empty(),
-                        rule.0.script.is_some(),
-                        rule.0.region.is_some(),
-                        Some(&rule.0.variants),
-                        &rule.1,
-                    );
-                    result = CanonicalizationResult::Modified;
-                    matched = true;
-                    break;
+                #[allow(clippy::unwrap_used)]
+                if let Ok(from) = rule.0.parse() {
+                    if uts35_rule_matches(locale, &from) {
```

Comment:
Suggestion: I think this is the UnparsedLanguageIdentifierPair table. Since you need to parse and match every LangaugeIdentifier in this table every time, you should ideally be using `cmp_bytes` so you don't need to parse and potentially allocate memory.

Reply:
We can't use `cmp_bytes`, but I've cleaned up the parsing